packages/core/core/src/utils/nuxt/_proxyUtils.ts

```diff
@@ -14,7 +14,7 @@ export const getBaseUrl = (req: IncomingMessage, basePath: string | undefined =
   const { headers } = req;
   const isHttps = require('is-https')(req);
   const scheme = isHttps ? 'https' : 'http';
-  const host = headers['x-forwarded-host'] || headers.host;
+  const host = process.env.API_HOST || headers['x-forwarded-host'] || headers.host;
```

Comment:
it would be nice, if we could change from direct access a `env` we can have a custom declaration in the settings file, and then use that as the detection

Reply:
if you pass this value in setting file then you won't be able to use the same codebase for different domains/subdomains where e.g. are different stores handled

Reply:
Gotacha!

So I don't see a problem.